internal/dot/graph.go

```diff
@@ -50,21 +50,45 @@ type Ctor struct {
 	ErrorType   ErrorType
 }
 
+func (c *Ctor) removeParams(r *Result) {
+	var pruned []*Param
+	for _, p := range c.Params {
+		if r.nodeKey() != p.nodeKey() {
+			pruned = append(pruned, p)
+		}
+	}
+	c.Params = pruned
+}
+
+type nodeKey struct {
+	t     reflect.Type
+	name  string
+	group string
+}
+
 // Node is a single node in a graph and is embedded into Params and Results.
 type Node struct {
 	Type  reflect.Type
 	Name  string
 	Group string
 }
 
-// Param is a parameter node in the graph.
+func (n *Node) nodeKey() nodeKey {
+	return nodeKey{n.Type, n.Name, n.Group}
```

Comment:
nit: prefer keyed struct literals

    nodeKey{t: .., name: .., group: ..}

Some linters will complain if you don't.